kotlin-coroutines-end/app/src/main/java/com/example/android/kotlincoroutines/main/RefreshMainDataWork.kt

```diff
@@ -18,44 +18,38 @@ package com.example.android.kotlincoroutines.main
 
 import android.content.Context
 import androidx.annotation.WorkerThread
+import androidx.work.CoroutineWorker
 import androidx.work.Worker
 import androidx.work.WorkerParameters
-import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.*
 
 /**
  * Worker job to refresh refresh titles from the network while the app is in the background.
  *
  * WorkManager is a library used to enqueue work that is guaranteed to execute after its constraints
  * are met. It can run work even when the app is in the background, or not running.
  */
-class RefreshMainDataWork(context: Context, params: WorkerParameters) : Worker(context, params) {
+class RefreshMainDataWork(context: Context, params: WorkerParameters) :
+        CoroutineWorker(context, params) {
+
+    override val coroutineContext = Dispatchers.IO
 
     /**
-     * Do our actual processing for the worker.
-     *
+     * Refresh the title from the network using [TitleRepository]
+     * 
      * WorkManager will call this method from a background thread. It may be called even
      * after our app has been terminated by the operating system, in which case [WorkManager] will
      * start just enough to run this [Worker].
      */
-    override fun doWork(): Result {
-        return refreshTitle()
-    }
-
-    /**
-     * Refresh the title from the network using [TitleRepository]
-     */
-    @WorkerThread
-    fun refreshTitle(): Result {
-        return runBlocking {
-            val database = getDatabase(applicationContext)
-            val repository = TitleRepository(MainNetworkImpl, database.titleDao)
+    override suspend fun doWork(): Result = coroutineScope {
```

Comment:
where is the coroutineScope coming from? could you update the comment mentioning this?

Reply:
Added a comment to explain that, by default, CoroutineWorker runs onDispatchers.Default; this can be modified by overriding coroutineContext.

Reply:
@florina-muntenescu coroutineScope is a function provided in the Coroutines Library https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/coroutine-scope.html

Reply:
This doesn't need a context since it is a suspend function that calls a suspend function.

CoroutineScope is only required to launch new coroutines. Suggest removing it to simplify the code.

Reply:
CoroutineWorker implementation uses CoroutineScope to launch a new coroutine, where doWork is executed.

[Here is the code in KTX](https://github.com/aosp-mirror/platform_frameworks_support/blob/4d7787e10d8a8637250f4e8f2b43ae1b3ccb2555/work/workmanager-ktx/src/main/java/androidx/work/CoroutineWorker.kt#L64). Here the idea is to launch This coroutine on the IO Dispatcher. 

Anyway, there is a launch behind the scene.

Reply:
I don't think you need `coroutineScope` here. This suspend function will be executed in the coroutine defined in the link you mentioned.

Reply:
I've removed the coroutine scope and updated the PR.